src/Build/Xml/XmlReaderExtension.cs

```diff
@@ -71,15 +83,61 @@ public void Dispose()
             _stream?.Dispose();
         }
 
+        /// <summary>
+        /// Returns <see cref="PropertyInfo"/> of the "Normalization" internal property on the given <see cref="XmlReader"/>-derived type.
+        /// </summary>
+        private static PropertyInfo GetNormalizationPropertyInfo(Type xmlReaderType)
```

Comment:
Do you need to pass this in as an argument? It's only called once, so you could replace its use with `typeof(XmlReader)`, right? It's private, so this doesn't really matter, but that would mean you could turn this into a property with a getter, which would be a little cleaner, and it would mean you couldn't accidentally call this with an invalid type, which would mean propertyInfo would be guaranteed not-null, and you could remove the debug else check below. (Keeping it isn't too bad, as it is good to know when your code is acting unexpectedly.) As a small side bonus, if you remove it, I don't think reader could be null, so you could replace `if (reader == null)` with `else`.

Reply:
Reading over your comments, is Normalization still not accessible on Core? That would make this hit the debug case fairly regularly, then, right?

Reply:
The object returned from `XmlReader.Create` is a BCL-internal subclass of `XmlReader`, so its type is different than `typeof(XmlReader)` and the property is not virtual.

`Normalization` is available on Core, Framework, and Mono, so on all supported runtimes this code works today. The debug/assert is there to catch the unlikely regression where a runtime refactors its XML library and the property disappears.